pennylane/transforms/qmc.py

```diff
@@ -322,34 +354,35 @@ def qmc():
          'diff_method': 'best',
          'gradient_fn': 'backprop'}
     """
+    operations = tape.operations.copy()
     wires = Wires(wires)
     target_wire = Wires(target_wire)
     estimation_wires = Wires(estimation_wires)
 
     if Wires.shared_wires([wires, estimation_wires]):
         raise ValueError("No wires can be shared between the wires and estimation_wires registers")
 
-    @wraps(fn)
-    def wrapper(*args, **kwargs):
-        fn(*args, **kwargs)
+    updated_operations = []
+    with qml.queuing.QueuingManager.stop_recording():
+        updated_operations.extend(operations)
         for i, control_wire in enumerate(estimation_wires):
-            Hadamard(control_wire)
+            updated_operations.append(Hadamard(control_wire))
 
             # Find wires eligible to be used as helper wires
             work_wires = estimation_wires.toset() - {control_wire}
             n_reps = 2 ** (len(estimation_wires) - (i + 1))
 
-            q = apply_controlled_Q(
-                fn,
+            tapes_q, fn = apply_controlled_Q(
```

Comment:
goes with below concern
```suggestion
            [tape_q], _ = apply_controlled_Q(
```